src/manipulators/strict_null_checks_manipulator.ts

```diff
@@ -26,9 +37,270 @@ export class StrictNullChecksManipulator extends Manipulator {
+              errorNode.getText() + '!'
+            );
+
+            const modifiedStatement = this.getModifiedStatement(newNode);
+
+            modifiedStatementedNodes.add([
+              (modifiedStatement.getParentOrThrow() as unknown) as StatementedNode,
+              modifiedStatement.getChildIndex(),
+            ]);
+          }
+          break;
+        }
+
+        // When two types are not assignable to each other, add both types to the declaration (variable,
+        // parameter, or property declaration)
+        case ErrorCodes.TypeANotAssignableToTypeB: {
+          if (!this.nodeKinds.has(errorNode.getKind())) {
+            return;
+          }
+
+          if (
+            Node.isIdentifier(errorNode) ||
+            Node.isPropertyAccessExpression(errorNode)
+          ) {
+            const errorSymbol = errorNode.getSymbolOrThrow();
+            const declarations = errorSymbol.getDeclarations();
+
+            // Determine the type that was assigned to the variable/parameter/property
+            const typesToAdd = this.determineAssignedType(errorNode);
+
+            // For each declaration, add the union of all declared and assigned types to modifiedDeclarationTypes
+            declarations.forEach(declaration => {
+              if (
+                Node.isVariableDeclaration(declaration) ||
+                Node.isParameterDeclaration(declaration) ||
+                Node.isPropertyDeclaration(declaration)
+              ) {
+                const declarationType = this.toTypeList(declaration.getType());
+
+                if (modifiedDeclarationTypes.has(declaration)) {
+                  declarationType.forEach(type => {
+                    modifiedDeclarationTypes.get(declaration)?.add(type);
+                  });
+                } else {
+                  modifiedDeclarationTypes.set(
+                    declaration,
+                    new Set(declarationType)
+                  );
+                }
+
+                typesToAdd.forEach(type => {
+                  modifiedDeclarationTypes.get(declaration)?.add(type);
+                });
+              }
+            });
+
+            // If error node is a return statement, add definite assignment assertion to return value
+          } else if (Node.isReturnStatement(errorNode)) {
+            if (
+              errorNode.getChildCount() === 1 ||
+              (errorNode.getChildCount() === 2 &&
+                errorNode.getLastChildOrThrow().getKind() ===
+                  SyntaxKind.SemicolonToken)
+            ) {
+              errorNode = errorNode.replaceWithText('return undefined!;');
+            } else {
+              errorNode = errorNode.replaceWithText(
+                'return (' + errorNode.getChildAtIndex(1).getText() + ')!;'
+              );
+            }
+
+            modifiedStatementedNodes.add([
+              (errorNode.getParentOrThrow() as unknown) as StatementedNode,
```

Comment:
Why do you need the double cast?

Reply:
I ran into the following error when I tried doing a direct cast:

`Conversion of type 'Node<Node>' to type 'StatementedNode' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.`

Seems like TypeScript has some extra guards for protecting against potentially dangerous casts. One way to get around this is assigning the parent node to a variable and doing a type guard, so I'll try doing that instead.